nomad/operator_endpoint.go

```diff
@@ -470,3 +473,155 @@ func (op *Operator) snapshotSave(conn io.ReadWriteCloser) {
 		}
 	}
 }
+
+func (op *Operator) snapshotRestore(conn io.ReadWriteCloser) {
+	defer conn.Close()
+
+	var args structs.SnapshotRestoreRequest
+	var reply structs.SnapshotRestoreResponse
+	decoder := codec.NewDecoder(conn, structs.MsgpackHandle)
+	encoder := codec.NewEncoder(conn, structs.MsgpackHandle)
+
+	handleFailure := func(code int, err error) {
+		encoder.Encode(&structs.SnapshotRestoreResponse{
+			ErrorCode: code,
+			ErrorMsg:  err.Error(),
+		})
+	}
+
+	if err := decoder.Decode(&args); err != nil {
+		handleFailure(500, err)
+		return
+	}
+
+	// Forward to appropriate region
+	if args.Region != op.srv.Region() {
+		err := op.forwardStreamingRPC(args.Region, "Operator.SnapshotRestore", args, conn)
```

Comment:
If we recorded a metric for operator bravery, we would increment it here. :sweat_smile: (Not that it shouldn't work fine... just that running a sensitive raft operation over the Internet is... brave.)